component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java

```diff
@@ -253,7 +255,49 @@ public Response createSpace(@Context UriInfo uriInfo,
 
     return EntityBuilder.getResponse(EntityBuilder.buildEntityFromSpace(space, authenticatedUser, uriInfo.getPath(), expand), uriInfo, RestUtils.getJsonMediaType(), Response.Status.OK);
   }
-  
+
+  /**
+   * {@inheritDoc}
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{spaceId}/checkExternals")
+  @RolesAllowed("users")
+  @ApiOperation(value = "Checks if a specific a space contains an external members",
+          httpMethod = "GET",
+          response = Response.class,
+          notes = "This returns the space if it contains external members")
+  @ApiResponses(value = {
+          @ApiResponse(code = 200, message = "Request fulfilled"),
+          @ApiResponse(code = 500, message = "Internal server error"),
+          @ApiResponse(code = 404, message = "Resource not found"),
+          @ApiResponse(code = 400, message = "Invalid query input")})
+  public Response isSpaceContainsExternals(@Context UriInfo uriInfo,
+                                           @Context Request request,
+                                           @ApiParam(value = "Space Id", required = true) @PathParam("spaceId") String spaceId) throws JSONException {
+    String authenticatedUser = ConversationState.getCurrent().getIdentity().getUserId();
+
+    if (spaceId == null) {
+      return Response.status(Status.BAD_REQUEST).entity("space Id is mandatory").build();
+    }
+
+    Space space = spaceService.getSpaceById(spaceId);
+    if (space == null) {
+      return Response.status(Status.NOT_FOUND).entity("space not found").build();
+    }
+    boolean hasExternals;
+    try {
+      hasExternals = spaceService.isSpaceContainsExternals(Long.valueOf(spaceId));
+    } catch (Exception e) {
+      LOG.error("Error while checking external members on space {}", space.getGroupId(), e);
+      return Response.status(Status.INTERNAL_SERVER_ERROR).entity("server internal error occurred").build();
+    }
+    JSONObject jsonObject = new JSONObject();
+    jsonObject.put("hasExternals", hasExternals);
+
+    return Response.ok(jsonObject.toString()).build();
```

Comment:
As what we do in ExoFeatureRest (https://github.com/Meeds-io/commons/blob/f702a1f9881630ca8d7021a5ba7d985d6c48afe9/commons-component-common/src/main/java/org/exoplatform/settings/rest/ExoFeatureRest.java#L33) when we need to return only true or false, can you return a text/pain, containing only the answer ? No need to use json

Reply:
Done